src/main/java/io/debezium/connector/db2/Db2StreamingChangeEventSource.java

```diff
@@ -325,6 +326,9 @@ private Db2ChangeTable[] getCdcTablesToQuery(Db2Partition partition, Db2OffsetCo
             }
             if (schema.tableFor(currentTable.getSourceTableId()) == null) {
                 LOGGER.info("Table {} is new to be monitored by capture instance {}", currentTable.getSourceTableId(), currentTable.getCaptureInstance());
+                // this prevents potential NPE if there is no TableId information in the source info
+                offsetContext.event(currentTable.getSourceTableId(), Instant.now());
+
```

Comment:
@jcechace I found that this was required for the test to pass, I'm not sure if this is indicative of a larger problem and that perhaps schema change events are being emitted with potentially incorrect table-id source info block values.

Reply:
@Naros I've just double checked SQL Server code and now they are in sync. It was in fact bug https://issues.redhat.com/browse/DBZ-2303